src/Resources/app/storefront/src/js/filter-slider-range.plugin.js

```diff
@@ -0,0 +1,321 @@
+import '../../node_modules/nouislider/distribute/nouislider.css';
+import deepmerge from 'deepmerge';
+import DomAccess from 'src/helper/dom-access.helper';
+import FilterBasePlugin from 'src/plugin/listing/filter-base.plugin';
+
+const noUiSlider = require('@nouislider');
+
+export default class FilterSliderRange extends FilterBasePlugin {
+
+  static options = deepmerge(FilterBasePlugin.options, {
+    inputMinSelector: '.min-input',
+    inputMaxSelector: '.max-input',
+    inputInvalidCLass: 'is-invalid',
+    inputTimeout: 500,
+    minKey: 'min-price',
+    maxKey: 'max-price',
+    price: {
+      min: 0,
+      max: 1,
+      step: 0.1
+    },
+    errorContainerClass: 'filter-range-error',
+    containerSelector: '.filter-range-container',
+    sliderContainer: '.fl--range-slider',
+    snippets: {
+      filterRangeActiveMinLabel: '',
+      filterRangeActiveMaxLabel: '',
+      filterRangeErrorMessage: ''
+    }
+  });
+
+  init() {
+    this._container = DomAccess.querySelector(this.el, this.options.containerSelector);
+    this._inputMin = DomAccess.querySelector(this.el, this.options.inputMinSelector);
+    this._inputMax = DomAccess.querySelector(this.el, this.options.inputMaxSelector);
+    this._timeout = null;
+    this._hasError = false;
+
+    this.slider = document.createElement('div');
+    this._sliderContainer = DomAccess.querySelector(this.el, this.options.sliderContainer);
+    this._sliderContainer.prepend(this.slider);
+
+    let start = this._inputMin.value.length ? this._inputMin.value : this.options.price.min;
+    let end = this._inputMax.value.length ? this._inputMax.value : this.options.price.max;
+
+    noUiSlider.create(this.slider, {
+      start: [start, end],
+      connect: true,
+      step: this.options.price.step,
+      range: {
+        'min': this.options.price.min,
+        'max': this.options.price.max,
+      },
+    });
+
+    this._registerEvents();
+  }
+
+  /**
+   * @private
+   */
+  _registerEvents() {
+    // Register slider events
+    this.slider.noUiSlider.on('update', this.onUpdateValues.bind(this));
+    this.slider.noUiSlider.on('end', this._onChangeInput.bind(this));
+
+    this._inputMin.addEventListener('blur', this._onChangeMin.bind(this));
+    this._inputMax.addEventListener('blur', this._onChangeMax.bind(this));
+
+    this._inputMin.addEventListener('keyup', this._onInput.bind(this));
+    this._inputMax.addEventListener('keyup', this._onInput.bind(this));
+  }
+
+  /**
+   * @return {Object}
+   * @public
+   */
+  getValues() {
+    const values = {};
+
+    this.validateMinInput();
+    this.validateMaxInput();
+
+    if (this.hasMinValueSet()) {
+      values[this.options.minKey] = this._inputMin.value;
+    }
+
+    if (this.hasMaxValueSet()) {
+      values[this.options.maxKey] = this._inputMax.value;
+    }
+
+    return values;
+  }
+
+  /**
+   * @param {Object} e
```

Comment:
The keyup event does not fire an `Object`, but a `KeyboardEvent`. See https://developer.mozilla.org/en-US/docs/Web/API/Document/keyup_event.
```suggestion
   * @param {KeyboardEvent} e
```